spark/src/main/java/org/apache/iceberg/actions/BaseAction.java

```diff
@@ -144,6 +123,32 @@ protected String metadataTableName(String tableName, MetadataTableType type) {
     return manifestDF.union(otherMetadataFileDF).union(manifestListDF);
   }
 
+  protected Dataset<Row> loadMetadataTable(SparkSession spark, String tableName, MetadataTableType type) {
+    String metadataTableName = metadataTableName(tableName, type);
```

Comment:
Maybe this method can not handle some corner cases.
For example, in the spark2 action, the file_path is very special (file://level1.level2.level3.level4/database/table).